### DISCUSSION TITLE
How to create usage charge?

### DISCUSSION THREAD

**Discussion by chintankotadia** [Kyon147/laravel-shopify#357] [General]
I would like to charge the store based on the number of orders placed per month if the store crosses the limit specified in recurring billing. 

For e.g., if the merchant has selected the basic recurring plan of $10 and it crosses the 1000 orders limit per month then I would like to charge `$10 Recurring fees + (100 more orders * 0.01$) = $11 `

It is mentioned in the wiki over [here](https://github.com/Kyon147/laravel-shopify/wiki/Creating-a-Billable-App#creating-usage-charges) but I would like to charge them instantly without redirecting them for approval. How can I do this one?

Thank you.

**Comment by chintankotadia** [comment]
@Kyon147 : Thoughts?

**Reply by Kyon147** [reply]
You cannot use the Shopify billing api and not as for approval. 

So you create the usage charge following the wiki and they have to approve it. 

Once approved you can bill them up to the amount in the charge.

**Reply by tariqbilal** [reply]
Hi @Kyon147,

I’m aiming to implement a similar functionality — creating usage charges where the user is automatically billed based on the services they use.

Currently, I’ve manually written the entire billing cycle, but it’s not producing ideal results. Since I’m already using this package, I realize it would be more efficient to handle it through the built-in approach.

Just to confirm my understanding:

Should I create a usage charge for the maximum amount, and the user will be automatically upgraded on the frontend (which I already manage)?

In my case, there are four packages, and users can resubscribe until they reach the highest package — they can also downgrade if needed.

Please let me know if I’m misunderstanding the recommended flow or missing a key step.

Thanks in advance for your guidance!
